fmt tool improvements

Zajozor opened the pull request:
Closes #898 
Sorry it took a while - but i needed to learn more than anticipated :)

Changes in this PR:
- **unrelated to fmt** - I own an M1 Mac and the builds of most of the creates were failing for me unless I updated neon. I am not sure about the impact on this on the rest of the project though. Let me know if it breaks something and I'll remove the commit.

- Adds tests to fmt, with some basic examples
- Fixes the original issue from #898 as well as some things about BinaryExpression/Expression formatting
- Some code cleanup
- Adds a README with a tutorial on how to get this running in your vscode

Overall fmt is still far form perfect but I see this as a step in a good direction.

btw sorry if some of my changes still refer to the old name, you literally renamed 3 hours ago 😅

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/slint-ui/slint?pullRequest=929) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/slint-ui/slint?pullRequest=929) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/slint-ui/slint?pullRequest=929) it.</sub>

Zajozor commented on tools/fmt/main.rs:
```diff
@@ -134,63 +136,47 @@ fn process_markdown_file(source: String, mut file: impl Write) -> std::io::Resul
+    match path.extension() {
+        Some(ext) if ext == "rs" => return process_rust_file(source, file),
+        Some(ext) if ext == "md" => return process_markdown_file(source, file),
+        Some(ext) if ext == "60" => return process_60_file(source, path, file),
```
so actually are the files now also called .slint ? i'll look into renaming these in the next days

ogoffart commented on tools/fmt/writer.rs:
```diff
@@ -0,0 +1,37 @@
+// Copyright © SixtyFPS GmbH <info@sixtyfps.io>
+// SPDX-License-Identifier: (GPL-3.0-only OR LicenseRef-SixtyFPS-commercial)
```
The CI system is checking the license header on every file, and that is the old one.

```suggestion
// Copyright © SixtyFPS GmbH <info@slint-ui.com>
// SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-commercial
```

ogoffart reviewed (commented):
Thank you very much. This look s like a great improvement

Yeah, sorry that you were disrupted by the naming change. There is still quite a lot of references to .60 and SixtyFPS, but i can as well fix them after the merge of this PR.

However let's just try to make the CI green before merging it.

Zajozor commented:
You were right with the node16 thing, i was using node17 before, that was the problem, didn't realize :)

I removed the commit which upgraded neon and also renamed all the references of sixtyfps to slint.

Let me know if there is something else that needs to be done before merging! (from my pov it's ready)

ogoffart commented:
Looks good! Thank you for the contribution
